chartfx-dataset/src/main/java/de/gsi/dataset/serializer/spi/BinarySerialiser.java

```diff
@@ -279,383 +313,397 @@ public static double getDouble(final IoBuffer readBuffer) {
+            }
+            keys = (K[]) new Object[nElements];
+            for (int i = 0; i < keys.length; i++) {
+                keys[i] = (K) serialiser.getReturnObjectFunction().apply(this, null, null);
+            }
+        }
+        // read value type and value vector
+        final V[] values;
+        final DataType valueDataType = getDataType(buffer.getByte());
+        if (valueDataType != DataType.OTHER) {
+            values = getGenericArrayAsBoxedPrimitive(valueDataType);
+        } else {
+            final String classTypeName = buffer.getStringISO8859();
+            final String secondaryTypeName = buffer.getStringISO8859();
+            final Type classType = ClassUtils.getClassByName(classTypeName);
+            final Type[] secondaryType = secondaryTypeName.isEmpty() ? new Type[0] : new Type[] { ClassUtils.getClassByName(secondaryTypeName) };
+            if (serialiserLookup == null) {
+                throw new IllegalArgumentException("protocol error: serialiserLookup must not be null for DataType == OTHER");
+            }
+            final FieldSerialiser serialiser = serialiserLookup.apply(classType, secondaryType);
+
+            if (serialiser == null) {
+                throw new IllegalArgumentException(NO_SERIALISER_IMP_FOUND + classTypeName);
+            }
+            values = (V[]) new Object[nElements];
+            for (int i = 0; i < values.length; i++) {
+                values[i] = (V) serialiser.getReturnObjectFunction().apply(this, null, null);
+            }
+        }
+
+        // generate new/write into existing Map
         final Map<K, V> retMap = map == null ? new ConcurrentHashMap<>() : map;
+        if (map != null) {
+            map.clear();
+        }
         for (int i = 0; i < keys.length; i++) {
-            retMap.put((K) keys[i], (V) values[i]);
+            retMap.put(keys[i], values[i]);
         }
 
         return retMap;
     }
 
-    protected static int getNumberOfElements(final int[] dimensions) {
-        AssertUtils.notNull("dimensions", dimensions);
-        int ret = 1;
-        for (int dim : dimensions) {
-            ret *= dim;
-        }
-        return ret;
+    @Override
+    public WireDataFieldDescription getParent() {
+        return parent;
     }
 
-    public static <E> Queue<E> getQueue(final IoBuffer readBuffer, final Queue<E> collection) {
-        final DataType valueDataType = DataType.fromByte(readBuffer.getByte());
-
-        // read value vector
-        final Object[] values = getGenericArrayAsBoxedPrimitive(readBuffer, valueDataType);
-        final int nElements = values.length;
-        final Queue<E> retCollection = collection == null ? new PriorityQueue<>(nElements) : collection;
-        for (int i = 0; i < nElements; i++) {
-            retCollection.add((E) values[i]);
+    @Override
+    public <E> Queue<E> getQueue(final Queue<E> collection, BiFunction<Type, Type[], FieldSerialiser<E>> serialiserLookup) {
```

Comment:
Codacy found an issue: [The method 'getQueue(Queue, BiFunction)' has an NPath complexity of 204, current threshold is 200](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5905529)